py3status/modules/xscreensaver.py

```diff
@@ -0,0 +1,91 @@
+# -*- coding: utf-8 -*-
+"""
+Toggle and lock Xscreensaver.
+
+This is a partial rewrite of dpms.py. Most credit goes to the original author
+Andre Doser. This module shows and controls the status of Xscreensaver
+
+This script is useful for people who let Xscreensaver manage DPMS settings.
+Xscreensaver has its own dpms variables separate from xset. DPMS can be
+safely turned off in xset as long as Xscreensaver is running.
+Settings can be managed using "xscreensaver-demo".
+
+Configuration parameters:
+    button_lock: mouse button to lock Xscreensaver (default 3)
+    button_toggle: mouse button to toggle Xscreensaver (default 1)
+    cache_timeout: refresh interval for this module (default 15)
+    format: default format for this module (default '{icon}')
+    icon_off: display when Xscreensaver is not running (default 'XSCR')
+    icon_on: display when Xscreensaver is running (default 'XSCR')
+
+ Format placeholders:
+    {icon} Xscreensaver icon
+
+@author neutronst4r <c7420{at}posteo{dot}net>
+"""
+from os import setpgrp
+from subprocess import Popen, DEVNULL
+STRING_UNAVAILABLE = "xscreensaver: isn't installed"
+
+
+class Py3status:
+    """
+    """
+    # available configuration parameters
+    button_lock = 3
+    button_toggle = 1
+    cache_timeout = 15
+    format = '{icon}'
+    icon_off = 'XSCR'
+    icon_on = 'XSCR'
+
+    def post_config_hook(self):
+        self.color_on = self.py3.COLOR_ON or self.py3.COLOR_GOOD
+        self.color_off = self.py3.COLOR_OFF or self.py3.COLOR_BAD
+
+    def _is_running(self):
+        try:
+            self.py3.command_output(['pidof', 'xscreensaver'])
+        except:
+            return False
+        else:
+            return True
+
+    def xscreensaver(self):
+        if not self.py3.check_commands(['xscreensaver']):
+            return {
+                'cached_until': self.py3.CACHE_FOREVER,
+                'color': self.py3.COLOR_BAD,
+                'full_text': STRING_UNAVAILABLE
+            }
```

Comment:
This should be done in `post_config_hook()` seeing as we have it defined.  By raising an `Exception(msg)` less lines too